ktor-client/ktor-client-core/common/src/io/ktor/client/plugins/sse/ClientSSESession.kt

```diff
@@ -16,7 +17,41 @@ public interface SSESession : CoroutineScope {
     /**
      * An incoming server-sent events flow.
      */
-    public val incoming: Flow<ServerSentEvent>
+    public val incoming: Flow<ServerSentEvent<String>>
+}
+
+/**
+ * A Server-sent events session.
+ */
+public interface SSESessionWithDeserialization : SSESession {
+    /**
+     * Deserializer for transforming field `data` of `ServerSentEvent` into desired data object.
+     */
+    public val deserializer: (TypeInfo) -> (String) -> Any
```

Comment:
Could you tell me why it's not just `fun deserialize(TypeInfo, String) -> Any`?

Reply:
Added `Deserializer` and `Serializer` interface:
```kotlin
public interface Deserializer {
    public fun deserialize(typeInfo: TypeInfo, data: String): Any
}
```
Do we want to make them functional because now it's used like:
```kotlin
client.sse(
    { url("$TEST_SERVER/sse/person") },
    object : Deserializer {
        override fun deserialize(typeInfo: TypeInfo, data: String): Any {
            return Person1(data)
    }
}) {
    incoming.single().apply {
        assertEquals("Name 0", deserialize<Person1>(data)?.name)
    }
}
```